src/components/ActionControl/ActionControl.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import classnames from "classnames";
+import React, { PropsWithChildren } from "react";
+import styles from "./ActionControl.module.css";
+
+import { Control, Field, Root } from "../Form";
+
+type ActionControlProps = {
+  className?: string;
+  Icon: unknown; // Don't know how to type this?
+  onActionClick: (e: React.MouseEvent) => void;
+  actionLabel: string;
+  disabled?: boolean;
+} & React.ComponentProps<typeof Control>;
+
+export const ActionControl = ({
+  children,
+  Icon,
+  className,
+  actionLabel,
+  onActionClick,
+  ...props
+}: PropsWithChildren<ActionControlProps>) => {
+  // TODO: Update the class name to something related to the component name
+  const classes = classnames(styles.actioncontrol, className);
+  // TODO: I don't know how to type this properly
+  const IconComp = Icon as typeof React.Component;
```

Comment:
Can you use the type from the '*.svg' module? `const ReactComponent: React.FunctionComponent<
    React.SVGAttributes<SVGElement>
  >;`

Reply:
Yup, that was the old way I had created icons.
Your suggestion is defo the way to go!